### PR TITLE
[castai-evictor] support custom root certificate authority

### PR DESCRIPTION
**Pull request by KarolisL**

### CODE REVIEW

File: charts/castai-evictor/templates/deployment.yaml [hunk 1]
```diff
@@ -92,6 +92,13 @@ spec:
             - configMapRef:
                 name: {{ include "evictor.name" . }}-config
                 optional: false
+            {{- if .Values.trustedCACert }}
+            - secretRef:
+                name: {{ include "castai-agent.fullname" . -}}-tls-cert
```
Comment (mikenorgate) [inline charts/castai-evictor/templates/deployment.yaml:97]: This should be `evictor.fullname` and you need to create the secret

File: charts/castai-evictor/templates/secret-tls-cert.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{{- if .Values.trustedCACert }}
```
Comment (saumas) [inline charts/castai-evictor/templates/secret-tls-cert.yaml:1]: Where is this `trustedCACert` coming from? It should be defined in `values.yaml`.

File: charts/castai-evictor/templates/secret-tls-cert.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "castai-agent.fullname" . }}-tls-cert
```
Comment (saumas) [inline charts/castai-evictor/templates/secret-tls-cert.yaml:6]: Should be `evictor.fullname`

File: charts/castai-evictor/templates/secret-tls-cert.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "castai-agent.labels" . | nindent 4 }}
```
Comment (saumas) [inline charts/castai-evictor/templates/secret-tls-cert.yaml:9]: `evictor.labels`

File: charts/castai-evictor/templates/secret-tls-cert.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  {{ if gt (len .Values.commonAnnotations) 0 -}}
+  annotations:
+    {{- include "castai-agent.annotations" . | nindent 4 }}
```
Comment (saumas) [inline charts/castai-evictor/templates/secret-tls-cert.yaml:12]: `evictor.annotations`

**Review by saumas — CHANGES_REQUESTED** [review]

File: charts/castai-evictor/templates/secret-tls-cert.yaml (hunk 2 above)
Comment (KarolisL) [reply charts/castai-evictor/templates/secret-tls-cert.yaml:1]: It is an optional value used by a small (currently, 1) subset of customers. Other services doesn't define it in `values.yaml` as well. Should I update those too, or just be consistent and don't define it anywhere?

**Review by saumas — APPROVED** [review]

**Pull request merged**